src/source/DataModelV2/LightingInstance.cpp

```diff
@@ -0,0 +1,260 @@
+{
+	if(strcmp(item->lpszPropName, "TopAmbient") == 0)
+	{
+		topAmbient = Color3(
+			GetRValue(item->lpCurValue)/255.0F,
+			GetGValue(item->lpCurValue)/255.0F,
+			GetBValue(item->lpCurValue)/255.0F
+		);
+	}
+	if(strcmp(item->lpszPropName, "BottomAmbient") == 0)
+	{
+		bottomAmbient = Color3(
+			GetRValue(item->lpCurValue)/255.0F,
+			GetGValue(item->lpCurValue)/255.0F,
+			GetBValue(item->lpCurValue)/255.0F
+		);
+	}
+	if(strcmp(item->lpszPropName, "SpotLight") == 0)
+	{
+		spotLight = Color3(
+			GetRValue(item->lpCurValue)/255.0F,
+			GetGValue(item->lpCurValue)/255.0F,
+			GetBValue(item->lpCurValue)/255.0F
+		);
+	}
+	if(strcmp(item->lpszPropName, "ClearColor") == 0)
+	{
+		clearColor = Color3(
+			GetRValue(item->lpCurValue)/255.0F,
+			GetGValue(item->lpCurValue)/255.0F,
+			GetBValue(item->lpCurValue)/255.0F
+		);
+	}
+	
+	else
+		Instance::PropUpdate(item);
+}
+
+
+// Functions
+void LightingInstance::suppressSky(bool doSuppress)
+{
+	_hideSky = doSuppress;
+}
+
+void LightingInstance::update()
+{
+	RenderDevice* rd = g_usableApp->getRenderDevice();
+
+    
+	if(!_hideSky) {
+		rd->clear(sky.isNull(), true, true);
+		if (sky.notNull()) sky->render(rd, lighting);
+	} else {
+		rd->setColorClearValue(clearColor);
+		rd->clear(true, true, true);
+		suppressSky(false);
+	}
+	
+    // Setup lighting
+    rd->enableLighting();
+
+	rd->setShadeMode(RenderDevice::SHADE_SMOOTH);
+	rd->setAmbientLightColor(Color3(1,1,1));
+
+	rd->setLight(0, GLight::directional(lighting.lightDirection, lighting.lightColor, true, true));
+	rd->setAmbientLightColor(lighting.ambient);
+
+	rd->beforePrimitive();
+	CoordinateFrame forDraw = rd->getObjectToWorldMatrix();
+	glEnableClientState(GL_VERTEX_ARRAY);
+	glEnableClientState(GL_COLOR_ARRAY);
+	glEnableClientState(GL_NORMAL_ARRAY);
+	g_dataModel->getWorkspace()->render(rd);
+	g_dataModel->getWorkspace()->renderName(rd);
+	glDisableClientState(GL_VERTEX_ARRAY);
+	glDisableClientState(GL_COLOR_ARRAY);
+	glDisableClientState(GL_NORMAL_ARRAY);
+	rd->setObjectToWorldMatrix(forDraw);
+	rd->afterPrimitive();
+	
+	// Draw outlines
+	for(size_t i = 0; i < g_dataModel->getSelectionService()->getSelection().size(); i++)
+	{
+		if(PartInstance* part = dynamic_cast<PartInstance*>(g_dataModel->getSelectionService()->getSelection()[i]))
+		{
+			Vector3 size = part->getSize();
+			Vector3 pos = part->getPosition();
+			drawOutlines(
+				Vector3(0+size.x/2, 0+size.y/2, 0+size.z/2) ,
+				Vector3(0-size.x/2,0-size.y/2,0-size.z/2), 
+				rd,
+				Vector3(size.x/2, size.y/2, size.z/2), 
+				Vector3(pos.x, pos.y, pos.z), 
+				part->getCFrame()
+			);
+		}		
+	}
+
+    rd->disableLighting();
+
+	drawEffects();
+}
+
+void LightingInstance::drawEffects()
+{
+	RenderDevice* rd = g_usableApp->getRenderDevice();
+    if (sky.notNull()) {
+        sky->renderLensFlare(rd, lighting);
+    }
+}
+
+void LightingInstance::drawOutlines(Vector3 from, Vector3 to, RenderDevice* rd, Vector3 size, Vector3 pos, CoordinateFrame c)
```

Comment:
This has nothing to do with lighting, it should be left alone for now; I have a future PR that will be replacing outlines